### PR TITLE
[-] FO : Fix #PSCSX-6751, gets the carrier available with weight attribute

### PR DESCRIPTION
**Pull request by webagil-kevin**
Considers the weight of products attributes.

Bug fix http://forge.prestashop.com/browse/PSCSX-6751

### CODE REVIEW

**Comment by gRoussac** [comment]
:gift_heart:

Thank you for your PR and dedication.

**Comment by gRoussac** [comment]
:+1:

**Comment by gRoussac** [comment]
Thank you for your PR and contribution.

Best regards

File: classes/Carrier.php [hunk 1]
```diff
@@ -1401,7 +1407,7 @@ public static function getAvailableCarrierList(Product $product, $id_warehouse,
                     unset($carrier_list[$key]);
                 }
 
-                if ($carrier->max_weight > 0 && $carrier->max_weight < $product->weight * $cart_quantity) {
+                if ($carrier->max_weight > 0 && ($carrier->max_weight < $product->weight * $cart_quantity OR $carrier->max_weight < $cart_weight)) {
```
Comment (nsorosac) [inline classes/Carrier.php:1410]: As mentionned here : http://forge.prestashop.com/browse/PSCSX-6751?focusedCommentId=156257&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-156257

Shouldn't this condition become only "if ($carrier->max_weight > 0 && $carrier->max_weight < $cart_weight)" ?

**Pull request merged**